odoo/addons/base/models/ir_ui_view.py

```diff
@@ -457,7 +457,7 @@ def _check_xml(self):
                     # fully upgraded already.
                     if self.pool._init and sibling_primary_views:
                         query = sibling_primary_views._get_filter_xmlid_query()
-                        sql = SQL(query, res_ids=tuple(sibling_primary_views.ids), modules=tuple(self.pool._init_modules))
+                        sql = SQL(query, res_ids=tuple(sibling_primary_views.ids), modules=tuple(self.pool._init_modules) + (self.env.context.get('install_module'),))
```

Comment:
```suggestion
                    if self.pool._init and sibling_primary_views and self.pool._init_modules:
                        query = sibling_primary_views._get_filter_xmlid_query()
                        sql = SQL(query, res_ids=tuple(sibling_primary_views.ids), modules=tuple(self.pool._init_modules))
```

Only the fully upgraded module is required. If you modify two templates in the same module there will be an error because the view is used.

E.g.: 
```xml
- <div t-attf-class="input-group #{_classes}" role="search">
+ <div t-attf-class="input-group #{_classes}" role="ooooo">
```
```xml
- <xpath expr="//div[@role='search']" position="attributes">
+ <xpath expr="//div[@role='ooooo']" position="attributes">
```